Rubberduck.Parsing/Symbols/IdentifierReferenceResolver.cs

```diff
@@ -196,6 +200,12 @@ private void ResolveLabel(ParserRuleContext context, string label)
                 var members = _declarationFinder.Members(module);
                 defaultMember = (IParameterizedDeclaration)members.FirstOrDefault(m => m is IParameterizedDeclaration && m.Attributes.HasDefaultMemberAttribute() && (isAssignmentTarget ? m.DeclarationType.HasFlag(DeclarationType.Procedure) : m.DeclarationType.HasFlag(DeclarationType.Function)));
             }
+            if (isRedimVariable)
+            {
+                // If the redim statement causes a variable to be created we have to add it manually to the state.      
+                // We rely on the fact that AddDeclaration removes/does not allow duplicates.
+                _state.AddDeclaration(boundExpression.ReferencedDeclaration);
```

Comment:
While the comment is good, I'm thinking that it might be better to make the behavior defined -- I would refactor the name `AddDeclaration` to `TryAddDeclaration` or even `TryAddNewDeclaration` to make it more obvious for future work that we are expecting the method to not add duplicates.